otter/convergence/service.py

```diff
@@ -799,3 +812,18 @@ def divergent_changed(self, children):
             # the return value is ignored, but we return this for testing
             eff = self._converge_all(my_buckets, children)
             return perform(self._dispatcher, self._with_conv_runid(eff))
+
+
+class ConvergenceExecutor(object):
+    gather = attr.ib()
+    plan = attr.ib()
+    get_desired_group_state = attr.ib()
+    update_cache = attr.ib()
+
+
+@attr.s
+class LaunchServerExecutor(ConvergenceExecutor):
```

Comment:
So the way these objects are done strikes me as a biiiit gratuitous, but I'm not really that bothered by it.

However, the thing that does seem weird to me, is that if we are going to use an @attr.s object, that the specialized versions should really just be instances of that class instead of a subclass that re-declares all the attributes with defaults.

So I see two options that to me would be slightly preferable to what this currently does

1. just use plain classes, where ConvergenceExecutor is an ABC, that declares the methods with no implementation, and two subclasses (LaunchServerExecutor and HeatExecutor) that implement those methods differently
2. continue to use objects as bags of methods, but put the `@attr.s` decorator on `ConvergenceExecutor`, and declare the actual implementations as instances instead of subclasses, e.g.:

```
launch_server_executor = ConvergenceExecutor(
    gather=get_all_launch_server_data,
    plan=plan_launch_server,
    ...)
```

What do you think?